app/src/main/java/org/simple/clinic/home/overdue/OverdueScreenController.kt

```diff
@@ -127,17 +126,17 @@ class OverdueScreenController @Inject constructor(
     return DAYS.between(date, LocalDate.now(UTC)).toInt()
   }
 
-  private fun phoneCallPermissionRequests(events: Observable<UiEvent>): Observable<UiChange> {
-    return events
-        .ofType<CallPatientClicked>()
-        .flatMapSingle { phoneNumberMaskerConfig }
-        .filter { it.showPhoneMaskBottomSheet.not() }
-        .map { { ui: Ui -> ui.requestCallPermission() } }
-  }
+  private fun phoneCallPermissionRequests(events: Observable<UiEvent>): Observable<UiChange> =
+      events
+          .withLatestFrom(phoneNumberMaskerConfig)
+          .filter { (_, config) -> config.showPhoneMaskBottomSheet.not() }
+          .map { (event, _) -> event }
```

Comment:
I know this has been removed and is no longer relevant, but this extra `.map` function can be avoided by moving the `.ofType()` call to the top of the chain (for future reference).

```kotlin
events
.ofType<CallPatientClicked>()
.withLatestFrom(phoneNumberMaskerConfig)
.filter { (_, config) -> config.showPhoneMaskBottomSheet.not() }
.map { { ui: Ui -> ui.requestCallPermission() } }
```